snuba/datasets/querylog.py

```diff
@@ -0,0 +1,86 @@
+from snuba import settings
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    Enum,
+    Float,
+    LowCardinality,
+    Mapping,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+)
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.querylog_processor import QuerylogProcessor
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.query.extensions import QueryExtension
+
+
+class QuerylogDataset(Dataset):
+    def __init__(self) -> None:
+        status_type = Enum([("success", 0), ("error", 1), ("rate-limited", 2)])
+
+        columns = ColumnSet(
+            [
+                ("request_id", UUID()),
+                ("request_body", String()),
+                ("referrer", LowCardinality(String())),
+                ("dataset", LowCardinality(String())),
+                ("projects", Array(UInt(64))),
+                ("organization", Nullable(UInt(64))),
+                ("timestamp", DateTime()),
+                ("duration_ms", UInt(32)),
+                ("status", status_type),
+                (
+                    "clickhouse_queries",
+                    Nested(
+                        [
+                            ("sql", String()),
```

Comment:
mind adding the query_id from stats. That's the cache key for deduplication and cache. It may be useful to see how many of a given query run into cache/dedupe